### PR TITLE
[Docs] Add and improve new api docs

### PR DESCRIPTION
**Pull request by AdamKasp**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| License         | MIT

### CODE REVIEW

File: docs/customization/api.rst [hunk 1]
```diff
@@ -0,0 +1,80 @@
+Customizing API
+===============
+
+For create all endpoints in Sylius API we are using the API Platform.
+API Platform allows to configure endpoint by ``yaml`` and ``xml`` files or in ``annotations``.
+In these guide we will show you how to customize Sylius API endpoint with ``xml``.
```
Comment (arti0090) [inline docs/customization/api.rst:6]:
```suggestion
In these guide we will show you how to customize Sylius API endpoint using ``xml`` configuration.
```

wdyt?

File: docs/customization/api.rst [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+How to prepare project to customization?
```
Comment (arti0090) [inline docs/customization/api.rst:8]:
```suggestion
How to prepare project for customization?
```

File: docs/customization/api.rst [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+========================================
+
+For any kind of customization firstly you need to configure location of your configs.
```
Comment (arti0090) [inline docs/customization/api.rst:11]: I would change `location` for `path` or `directory`. Location seems more suitable for navigation 😄

File: docs/customization/api.rst [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        json: ['application/merge-patch+json']
+    swagger:
+        versions: [3]
+
+And now if you want to customize any API resource you need copy entire configuration of this resource from ``%kernel.project_dir%/vendor/sylius/sylius/src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/`` to ``%kernel.project_dir%/config/api_platform``
+
+How to add an additional endpoint to a resource?
+================================================
+
+Let's assume that you want to add new endpoint to ``order`` which will dispatch a command.
```
Comment (arti0090) [inline docs/customization/api.rst:30]:
```suggestion
Let's assume that you want to add new endpoint to ``order`` resource which will dispatch a command.
```

File: docs/customization/api.rst [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+You need to add proper configuration in ``config/api_platform/Order.xml``:
+
+.. code-block:: xml
+
+    <collectionOperations>
+        <collectionOperation name="custom_operation">
+            <attribute name="method">POST</attribute>
+            <attribute name="path">/shop/order/custom-operation</attribute>
```
Comment (arti0090) [inline docs/customization/api.rst:38]:
```suggestion
            <attribute name="path">/shop/orders/custom-operation</attribute>
```

Well it is example and user can make it `order`, but we are using `orders` so I would stay with it 😄

File: docs/customization/api.rst [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+And that's all, now you have new endpoint with your custom logic.
+
+.. tip::
+
+    Read more about API Platform endpoint configuration `here <https://api-platform.com/docs/core/operations/>`_
+
+How to remove a resource endpoint?
+==================================
+
+Let's assume that your shop offer only digital products so you don't need setting shipment method at all.
```
Comment (arti0090) [inline docs/customization/api.rst:53]:
```suggestion
Let's assume that your shop offers only digital products so you don't need to setup shipment method at all.
```

File: docs/customization/api.rst [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+To remove endpoint you only need to delete unnecessary configuration from ``config/api_platform/Order.xml``
```
Comment (arti0090) [inline docs/customization/api.rst:54]: Is this sentence true? I mean we have some underlying logic that changes states of cart and checkout. Speaking of endpoint only POV, well we can just remove the endpoint and go drink some margarita 🍸 , but I don't think we can finish checkout process 🙁

File: docs/customization/api.rst [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+How to rename the resource endpoint's path?
+===========================================
+
+If you want to change endpoint's path, only what you need to do is change ``path`` attribute in your config:
```
Comment (arti0090) [inline docs/customization/api.rst:59]:
```suggestion
If you want to change endpoint's path, you need to change the ``path`` attribute in your config only:
```

File: docs/customization/api.rst [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+            <itemOperation name="admin_get">
+                <attribute name="method">GET</attribute>
+                <attribute name="path">/admin/orders/renamed-path/{id}</attribute>
+            </itemOperation>
+    </itemOperations>
+
+How to modify the endpoints prefixes? "shop" to "retail" rename.
+===============================================================
+
+Let's assume that you want to have your own prefixes on paths (for example to be more consistent with rest of your application)
```
Comment (arti0090) [inline docs/customization/api.rst:73]:
```suggestion
Let's assume that you want to have your own prefixes on paths (for example to be more consistent with rest of your application).
```

File: docs/customization/api.rst [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+As a first step you need to change ``paths`` or ``route_prefix`` attribute in all needed resources.
+Next step is modify security configuration in ``config/packages/security.yaml``, you need to overwrite parameter:
```
Comment (arti0090) [inline docs/customization/api.rst:75]:
```suggestion
Next step is to modify security configuration in ``config/packages/security.yaml``, you need to overwrite parameter:
```

File: docs/customization/api.rst (hunk 7 above)
Comment (AdamKasp) [reply docs/customization/api.rst:54]: yes, but actually no :D
if you have only one shipping method and enabled option `Skip shipping step if only one shipping method is available?` on the channel, while checkout you don't need to set shipping method :)
you can check how it works on UI :)

File: docs/customization/api.rst [hunk 11]
```diff
@@ -0,0 +1,80 @@
+Customizing API
+===============
+
+For create all endpoints in Sylius API we are using the API Platform.
```
[the remark this answers is not in the document]
Comment (CoderMaggie) [reply docs/customization/api.rst:4]: `We are using...` +1

File: docs/api/unified_api/sylius_api_paths.rst [hunk 12]
```diff
@@ -1,7 +1,8 @@
-Custom Paths in new Api
-=======================
+Sylius API paths
+================
 
 All paths in new API have the same prefix structure: ``/api/v2/admin/`` or ``/api/v2/shop/``
+Part ``/api/v2`` indicates the API version and ``/admin/`` or ``/shop/`` is necessary for authorization purpose.
```
Comment (CoderMaggie) [inline docs/api/unified_api/sylius_api_paths.rst:5]:
```suggestion
The ``/api/v2`` prefix part indicates the API version and the ``/admin/`` or ``/shop/`` prefixes are necessary for authorization purposes.
```

File: docs/customization/api.rst [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+API Platform allows to configure endpoint by ``yaml`` and ``xml`` files or in ``annotations``.
```
Comment (CoderMaggie) [inline docs/customization/api.rst:5]:
```suggestion
API Platform allows configuring an endpoint by ``yaml`` and ``xml`` files or by annotations.
```

File: docs/customization/api.rst (hunk 1 above)
Comment (CoderMaggie) [inline docs/customization/api.rst:6]:
```suggestion
In this guide, you will learn how to customize Sylius API endpoints using ``xml``.
```

File: docs/customization/api.rst [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+    mapping:
+        paths:
+            - '%kernel.project_dir%/vendor/sylius/sylius/src/Sylius/Bundle/ApiBundle/Resources/config/api_resources'
+            - '%kernel.project_dir%/config/api_platform'
+    patch_formats:
+        json: ['application/merge-patch+json']
+    swagger:
+        versions: [3]
+
+And now if you want to customize any API resource you need copy entire configuration of this resource from ``%kernel.project_dir%/vendor/sylius/sylius/src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/`` to ``%kernel.project_dir%/config/api_platform``
```
Comment (CoderMaggie) [inline docs/customization/api.rst:25]:
```suggestion
And now if you want to customize any API resource you need to copy the entire configuration of this resource from 
``%kernel.project_dir%/vendor/sylius/sylius/src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/`` to ``%kernel.project_dir%/config/api_platform``
```

File: docs/customization/api.rst (hunk 4 above)
Comment (CoderMaggie) [inline docs/customization/api.rst:30]:
```suggestion
Let's assume that you want to add a new endpoint to the ``Order`` resource that will be dispatching a command.
```

File: docs/customization/api.rst [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+    <collectionOperations>
+        <collectionOperation name="custom_operation">
+            <attribute name="method">POST</attribute>
+            <attribute name="path">/shop/order/custom-operation</attribute>
+            <attribute name="messenger">input</attribute>
+            <attribute name="input">App\Command\CustomCommand</attribute>
+        </collectionOperation>
+    </collectionOperations>
+
+And that's all, now you have new endpoint with your custom logic.
```
Comment (CoderMaggie) [inline docs/customization/api.rst:44]:
```suggestion
And that's all, now you have a new endpoint with your custom logic.
```

File: docs/customization/api.rst [hunk 16]
```diff
@@ -0,0 +1,80 @@
...
+
+.. tip::
+
+    Read more about API Platform endpoint configuration `here <https://api-platform.com/docs/core/operations/>`_
+
+How to remove a resource endpoint?
```
Comment (CoderMaggie) [inline docs/customization/api.rst:50]:
```suggestion
How to remove an endpoint?
```

File: docs/customization/api.rst [hunk 17]
```diff
@@ -0,0 +1,80 @@
...
+            - '%kernel.project_dir%/vendor/sylius/sylius/src/Sylius/Bundle/ApiBundle/Resources/config/api_resources'
+            - '%kernel.project_dir%/config/api_platform'
+    patch_formats:
+        json: ['application/merge-patch+json']
+    swagger:
+        versions: [3]
+
+And now if you want to customize any API resource you need copy entire configuration of this resource from ``%kernel.project_dir%/vendor/sylius/sylius/src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/`` to ``%kernel.project_dir%/config/api_platform``
+
+How to add an additional endpoint to a resource?
```
Comment (CoderMaggie) [inline docs/customization/api.rst:27]:
```suggestion
How to add an additional endpoint?
```

File: docs/customization/api.rst (hunk 6 above)
Comment (CoderMaggie) [inline docs/customization/api.rst:53]:
```suggestion
Let's assume that your shop is offering only digital products. Therefore, while checking out, 
your customers do not need to choose a shipping method for their orders. 

Thus you will need to modify the configuration file of the ``Order`` resource and remove the shipping method choosing endpoint from it.
```

File: docs/customization/api.rst (hunk 7 above)
Comment (CoderMaggie) [inline docs/customization/api.rst:54]:
```suggestion
To remove the endpoint you only need to delete the unnecessary configuration from your
``config/api_platform/Order.xml`` which is a copied configuration file, that overwrites the one from Sylius.
```

File: docs/customization/api.rst [hunk 18]
```diff
@@ -0,0 +1,80 @@
...
+.. tip::
+
+    Read more about API Platform endpoint configuration `here <https://api-platform.com/docs/core/operations/>`_
+
+How to remove a resource endpoint?
+==================================
+
+Let's assume that your shop offer only digital products so you don't need setting shipment method at all.
+To remove endpoint you only need to delete unnecessary configuration from ``config/api_platform/Order.xml``
+
```
Comment (CoderMaggie) [inline docs/customization/api.rst:55]: I think you could add a snippet of which lines have to be removed here. Or how the whole file would look like with the config removed, if it's not too long, or you can show just the neighboring lines of the section you'd be removing.

File: docs/customization/api.rst [hunk 19]
```diff
@@ -0,0 +1,80 @@
...
+How to rename the resource endpoint's path?
```
Comment (CoderMaggie) [inline docs/customization/api.rst:56]:
```suggestion
How to rename an endpoint's path?
```

File: docs/customization/api.rst (hunk 8 above)
Comment (CoderMaggie) [inline docs/customization/api.rst:59]:
```suggestion
If you want to change an endpoint's path, you just need to change the ``path`` attribute in your config:
```

File: docs/customization/api.rst [hunk 20]
```diff
@@ -0,0 +1,80 @@
...
+.. code-block:: xml
+
+    <itemOperations>
+            <itemOperation name="admin_get">
+                <attribute name="method">GET</attribute>
+                <attribute name="path">/admin/orders/renamed-path/{id}</attribute>
+            </itemOperation>
+    </itemOperations>
+
+How to modify the endpoints prefixes? "shop" to "retail" rename.
```
Comment (CoderMaggie) [inline docs/customization/api.rst:70]:
```suggestion
How to modify the endpoints prefixes?
```

File: docs/customization/api.rst (hunk 9 above)
Comment (CoderMaggie) [inline docs/customization/api.rst:73]:
```suggestion
Let's assume that you want to have your own prefixes on paths (for example to be more consistent with the rest of your application).
```

File: docs/customization/api.rst [hunk 21]
```diff
@@ -0,0 +1,80 @@
...
+===============================================================
+
+Let's assume that you want to have your own prefixes on paths (for example to be more consistent with rest of your application)
+As a first step you need to change ``paths`` or ``route_prefix`` attribute in all needed resources.
```
Comment (CoderMaggie) [inline docs/customization/api.rst:74]:
```suggestion
As the first step you need to change the ``paths`` or ``route_prefix`` attribute in all needed resources.
```
which are the `needed resources`? why not all?

File: docs/customization/api.rst (hunk 10 above)
Comment (CoderMaggie) [inline docs/customization/api.rst:75]:
```suggestion
The next step is to modify the security configuration in ``config/packages/security.yaml``, you need to overwrite the parameter:
```

File: docs/customization/api.rst [hunk 22]
```diff
@@ -0,0 +1,80 @@
...
+Next step is modify security configuration in ``config/packages/security.yaml``, you need to overwrite parameter:
+
+.. code-block:: xml
+
+    parameters:
+        sylius.security.new_api_shop_route: "%sylius.security.new_api_route%/retail"
```
Comment (CoderMaggie) [inline docs/customization/api.rst:80]: Any ending sentence? :D

File: docs/customization/api.rst [hunk 23]
```diff
@@ -0,0 +1,96 @@
+Customizing API
+===============
+
+We are using the API Platform to create all endpoints in Sylius API.
+API Platform allows configuring an endpoint by ``yaml`` and ``xml`` files or by annotations.
+In this guide, you will learn how to customize Sylius API endpoints using ``xml`` configuration.
+
+How to prepare project for customization?
+========================================
```
Comment (GSadee) [inline docs/customization/api.rst:9]:
```suggestion
=========================================
```

File: docs/customization/api.rst [hunk 24]
```diff
@@ -0,0 +1,96 @@
...
+
+.. code-block:: xml
+
+    <!-- delete this configuration -->
+    <itemOperation name="shop_select_shipping_method">
+        <!-- ... -->
+    </itemOperation>
+
+How to rename an endpoint's path?
+===========================================
```
Comment (GSadee) [inline docs/customization/api.rst:67]:
```suggestion
=================================
```

File: docs/customization/api.rst [hunk 25]
```diff
@@ -0,0 +1,96 @@
...
+The next step is to modify the security configuration in ``config/packages/security.yaml``, you need to overwrite the parameter:
+
+.. code-block:: xml
+
+    parameters:
+        sylius.security.new_api_shop_route: "%sylius.security.new_api_route%/retail"
+
+.. warning::
+
+    Changing prefixed without security configuration update can expose confidential data (like customers addresses)
```
Comment (GSadee) [inline docs/customization/api.rst:94]:
```suggestion
    Changing prefix without security configuration update can expose confidential data (like customers addresses)
```
?

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thanks, Adam! :1st_place_medal:

File: docs/customization/api.rst [hunk 26]
```diff
@@ -0,0 +1,97 @@
...
+    # config/packages/api_platform.yaml
+    api_platform:
+    mapping:
+        paths:
+            - '%kernel.project_dir%/vendor/sylius/sylius/src/Sylius/Bundle/ApiBundle/Resources/config/api_resources'
+            - '%kernel.project_dir%/config/api_platform'
+    patch_formats:
+        json: ['application/merge-patch+json']
+    swagger:
+        versions: [3]
```
Comment (lchrusciel) [inline docs/customization/api.rst:24]: Can we provide this config in Sylius Standard?

**Pull request merged**